SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -354,6 +375,151 @@ void InterestFactory::AddServerGameplayDebuggerCategoryReplicatorActorInterest(I
 }
 #endif
 
+bool UnrealServerInterestFactory::CreateClientInterestDiff(const APlayerController* PlayerController,
+														   ChangeInterestRequest& ChangeInterestRequestData, const bool bOverwrite) const
+{
+	USpatialNetConnection* NetConnection =
+		(PlayerController != nullptr) ? Cast<USpatialNetConnection>(PlayerController->NetConnection) : nullptr;
+	if (!ensure(GetDefault<USpatialGDKSettings>()->bUseClientEntityInterestQueries)
+		|| !ensureMsgf(PlayerController != nullptr, TEXT("Only PCs can use client entity interest: %s"), *GetNameSafe(PlayerController))
+		|| !ensureMsgf(NetConnection != nullptr, TEXT("Failed to retieve USpatialNetConnection from controller: %s"),
+					   *GetNameSafe(PlayerController)))
+	{
+		return false;
+	}
+
+	AActor* Pawn = PlayerController->GetPawn();
+	if (Pawn != nullptr && Pawn != PlayerController->GetViewTarget())
+	{
+		const_cast<APlayerController*>(PlayerController)->SetViewTarget(Pawn);
+	}
+	if (NetConnection->ViewTarget == nullptr)
+	{
+		NetConnection->ViewTarget = NetConnection->PlayerController->GetViewTarget();
+	}
+
+	TArray<Worker_EntityId> ClientInterestedEntities = GetClientInterestedEntityIds(PlayerController);
+
+	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(PlayerController->GetNetDriver());
+
+#if WITH_EDITOR
+	// Make debugging easier
+	ClientInterestedEntities.Sort();
+#endif
+
+	ChangeInterestRequestData.Clear();
+
+	{
+		// Add non-auth interest
+		TSet<Worker_EntityId_Key> FullInterested;
+		FullInterested.Reserve(ClientInterestedEntities.Num());
+		for (auto EntityId : ClientInterestedEntities)
+		{
+			FullInterested.Add(EntityId);
+		}
```

Comment:
```suggestion
		TSet<Worker_EntityId_Key> FullInterested(ClientInterestedEntities);
```
There is a constructor for that : )

Reply:
Hah, yeah this is intentional, unfortunately moving between Worker_EntityId and Worker_EntityId_Key.